src/pg_query_summary_statement_type.c

```diff
@@ -0,0 +1,798 @@
+#include "pg_query.h"
+#include "pg_query_internal.h"
+#include "pg_query_outfuncs.h"
+
+#include "parser/parser.h"
+#include "parser/scanner.h"
+#include "parser/scansup.h"
+
+#include "nodes/nodeFuncs.h"
+#include "utils/builtins.h"
+
+#include <unistd.h>
+#include <fcntl.h>
+#include <string.h>
+
+#include "pg_query_summary.h"
+
+// Declare the huge function early so we can put it at the end of the file.
+bool pg_query_summary_stmt_walk_impl(Node* node, Summary* summary);
```

Comment:
I think this function could use a `static` (both in this early declaration and the later definition)